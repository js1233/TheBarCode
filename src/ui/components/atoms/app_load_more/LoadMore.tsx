import { usePreferredTheme } from "hooks";
import React from "react";
import {
  ActivityIndicator,
  StyleProp,
  StyleSheet,
  TextStyle,
  TouchableOpacityProps,
  View
} from "react-native";
import { AppLabel } from "ui/components/atoms/app_label/AppLabel";

export interface AppLoadMoreProps extends TouchableOpacityProps {
  text?: string;
  loaderSize?: number;
  textStyle?: StyleProp<TextStyle>;
  shouldShowLoadMore?: boolean;
  loaderColor?: string;
}

export const LoadMore = React.memo<AppLoadMoreProps>(
  ({
    loaderSize = 25,
    text = "Loading, please wait",
    textStyle,
    loaderColor,
    shouldShowLoadMore = true
  }) => {
    const theme = usePreferredTheme();
    const view = () => {
      return (
        <>
          <ActivityIndicator
            size={loaderSize}
            color={loaderColor ?? theme.themedColors.interface["900"]}
          />
          <AppLabel
            text={text}
            style={[style.textStyle, textStyle]}
            numberOfLines={3}
          />
        </>
      );
    };
    return (
      <View style={style.container}>
        {shouldShowLoadMore ? view() : null}
      </View>
    );
  }
);

const style = StyleSheet.create({
  container: {
    width: "100%",
    alignItems: "center",
    justifyContent: "center",
    flexDirection: "row"
  },
  textStyle: {
    paddingLeft: 5
  }
});
