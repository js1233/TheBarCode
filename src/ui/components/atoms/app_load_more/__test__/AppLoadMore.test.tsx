import { render } from "@testing-library/react-native";
import React from "react";
import { LoadMore } from "ui/components/atoms/app_load_more/LoadMore";

test("snapshot testing", () => {
  const rendered = render(<LoadMore />).toJSON();
  expect(rendered).toMatchSnapshot();
});
