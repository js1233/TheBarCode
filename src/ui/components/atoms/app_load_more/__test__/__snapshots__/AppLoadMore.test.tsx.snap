// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <ActivityIndicator
    color="#171717"
    size={25}
  />
  <Text
    numberOfLines={3}
    style={
      Array [
        Object {
          "fontFamily": "ProximaNova-Regular",
        },
        Object {
          "color": "#171717",
          "fontSize": 27,
        },
        Object {
          "lineHeight": 42,
        },
        Array [
          Object {
            "paddingLeft": 5,
          },
          undefined,
        ],
      ]
    }
  >
    Loading, please wait
  </Text>
</View>
`;
